src/gallium/drivers/lima/lima_draw.c

```diff
@@ -824,6 +824,18 @@ lima_update_pp_fs_program(struct lima_context *ctx)
    ctx->buffer_state[lima_ctx_buff_pp_fs_program].size = align(fs->shader_size, 0x40);
 }
 
+static bool
+lima_is_scissor_zero(struct lima_context *ctx)
+{
+   if (!(ctx->dirty & LIMA_CONTEXT_DIRTY_SCISSOR))
```

Comment:
Like the previous commit, scissor may be zero in the second frame even not dirty.